Update basic send flow

rabbitholiness opened the pull request:
This PR addresses the follow-up tasks for the basic send flow page that are mentioned in [this issue](https://github.com/BitcoinDesign/Bitcoin-Core-App/issues/143).
More specifically, it:
- Updates the Basic Send Flow page
- Adds a page about the send form input fields, including fee selection. It covers the general approach to inline validation error handling and includes specific examples.

rabbitholiness commented:
@GBKS I think this is ready for a first review.

GBKS commented on assets/images/send/layout-blocks-amount-big.png:
Can you please make a couple of tweaks:
- Little bit of padding for "Send all"
- Make the input field fill the available space so it can be easily tapped to focus.

Would also be nice to be able to tap the available balance in the error message, to "Send all". That might be what the user wants to do.

GBKS replied:
If the label field is a fixed width, we may need to define that as well. That could get a little tricky across input fields, we the text size is dynamic and user-adjustable.

GBKS commented on pages/send/send-form-inputs.md:
```diff
@@ -38,6 +38,15 @@ Inline validation checks include:
+	image = "/assets/images/send/layout-blocks-address.png"
+	retina = "/assets/images/send/layout-blocks-address@2x.png"
+	big = "/assets/images/send/layout-blocks-address-big.png"
+	alt-text = "Layout blocks of the address input component."
```
Maybe just throw the image into AI and have it create alt text?

rabbitholiness commented:
Addressed @GBKS's comments.

GBKS commented:
Awesome progress. Just a few more questions & nits:

1. The offline banner is very aggressive visually. How about moving it inline, right above the buttons and making it more subtle? Instead of dead-ending users, could we tell them to go online, and give an option to save the transaction (as PSBT) until their next session?
2. Can we remove the cancel button in the review screen? Does not seem to add anything.
3. For the collapsed address, can we just tap it to expand it rather than having a tooltip? Could be just something we do without indication. Or maybe the three dots can be orange to indicate interaction?
5. Address highlighting: The darker parts are too dark and don't have enough contrast. It should not look like placeholder text.
6. “Send all” would be better worded as  “Send max” (the maximum amount including fees). My main concern is that this takes a lot of prime real estate for not being something people do all the time. Maybe it could just be one of the dropdown options? Send max really does two things, it adds the balance as the amount and it enables that fees are in the amount.
7. Insufficient balance error. We could also show "Send max" in here. "Your balance is XYZ. <a href="#">Send max</a>."
8. In the clipboard indicator, is “Paste” the right word? Maybe "Add" or "Insert". I'd also move the primary button to the right.

rabbitholiness commented:
@GBKS thanks for the feedback. Here we go: 

> The offline banner is very aggressive visually. How about moving it inline, right above the buttons and making it more subtle? Instead of dead-ending users, could we tell them to go online, and give an option to save the transaction (as PSBT) until their next session?

Changed it to blue

> Can we remove the cancel button in the review screen? Does not seem to add anything.

Done

> Address highlighting: The darker parts are too dark and don't have enough contrast. It should not look like placeholder text.

Changed from Neutral 5 to Neutral 6

> “Send all” would be better worded as “Send max” (the maximum amount including fees). My main concern is that this takes a lot of prime real estate for not being something people do all the time. Maybe it could just be one of the dropdown options? Send max really does two things, it adds the balance as the amount and it enables that fees are in the amount.

Agree, let's move it to the menu.

> Insufficient balance error. We could also show "Send max" in here. "Your balance is XYZ. [Send max](https://github.com/BitcoinDesign/Bitcoin-Core-App/pull/152#)."

Done

> In the clipboard indicator, is “Paste” the right word? Maybe "Add" or "Insert". I'd also move the primary button to the right.

Good point. I used "Fill", but "Insert" would also be good, I think. 

> For the collapsed address, can we just tap it to expand it rather than having a tooltip? Could be just something we do without indication. Or maybe the three dots can be orange to indicate interaction?

Hmm, I think it could be displayed as a tooltip on hover without the "info" icon. And on click the address could get copied to the clipboard. Not really sold about the orange three dots. wdyt?

GBKS reviewed (approved):
ACK. This is a good step forward. Still some minor stuff we can figure out in extra PRs. Let's get this chunk in.
